src/main/java/codeu/model/store/basic/MentionStore.java

```diff
@@ -0,0 +1,133 @@
+
+import codeu.model.data.Mention;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.Comparator;
+import java.util.Collections;
+
+/**
+ * Store class that uses in-memory data structures to hold values and automatically loads from and
+ * saves to PersistentStorageAgent. It's a singleton so all servlet classes can access the same
+ * instance.
+ */
+public class MentionStore {
+
+  /** Singleton instance of MentionStore. */
+  private static MentionStore instance;
+
+  /** The in-memory list of Mentions. */
+  private List<Mention> mentions;
+
+  /**
+   * Returns the singleton instance of MentionStore that should be shared between all servlet classes.
+   * Do not call this function from a test; use getTestInstance() instead.
+   */
+  public static MentionStore getInstance() {
+    if (instance == null) {
+      instance = new MentionStore(PersistentStorageAgent.getInstance());
+    }
+    return instance;
+  }
+
+  /**
+   * Instance getter function used for testing. Supply a mock for PersistentStorageAgent.
+   *
+   * @param persistentStorageAgent a mock used for testing
+   */
+  public static MentionStore getTestInstance(PersistentStorageAgent persistentStorageAgent) {
+    return new MentionStore(persistentStorageAgent);
+  }
+
+  /**
+   * The PersistentStorageAgent responsible for loading Mentions from and saving Mentions to Datastore.
+   */
+  private PersistentStorageAgent persistentStorageAgent;
+
+  /** This class is a singleton, so its constructor is private. Call getInstance() instead. */
+  private MentionStore(PersistentStorageAgent persistentStorageAgent) {
+    this.persistentStorageAgent = persistentStorageAgent;
+    mentions = new ArrayList<>();
+  }
+
+  /**
+   * Access the Mention object with the given name.
+   * @return null if username does not match any existing Mention
+   */
+  public Mention getMentionByName(String name) {
+    // This approach will be pretty slow if we have many mentions.
+    for (Mention mention : mentions) {
+      if (mention.getMentionedUser().equals(name)) {
+        return mention;
+      }
+    }
+    return null;
+  }
+
+  /** Adds new mention to current set of mentions.*/
+  public void addMention(Mention mention) {
+    if (isPresent(mention.getMentionedUser())){
+      return;
+    }
+    mentions.add(mention);
+    persistentStorageAgent.writeThrough(mention);
+  }
+
+/** Updates existing mention. */ 
+  public void updateMention(Mention mention) {
+    if (isPresent(mention.getMentionedUser()))  {
+      persistentStorageAgent.writeThrough(mention);
+    } 
+  } 
+
+  /**
+   * Access the  object with the given UUID.
+   *
+   * @return null if the UUID does not match any existing Mention.
+   */
+  public Mention getMentionById(UUID id) {
+    for (Mention mention : mentions) {
+      if (mention.getMessageIds().equals(id)) {
```

Comment:
This doesn't make sense, you are comparing a Set<UUID> to a single UUID. Mentions don't have a UUID property, what are you trying to do with this method?